russstation/code/modules/clothing/dwarf_clothes.dm

```diff
@@ -15,16 +25,74 @@
 	worn_icon = 'russstation/icons/mob/hands.dmi'
 	icon = 'russstation/icons/obj/clothing/gloves.dmi'
 	icon_state = "dwarf"
-	inhand_icon_state = "dwarf"
+	inhand_icon_state = "bgloves"
 	body_parts_covered = ARMS
 	species_exception = list(/datum/species/dwarf)
 
 /obj/item/clothing/shoes/dwarf
 	name = "dwarven shoes"
-	desc = "Standered issue dwarven mining shoes."
+	desc = "Standard issue dwarven mining shoes."
 	worn_icon = 'russstation/icons/mob/feet.dmi'
 	icon = 'russstation/icons/obj/clothing/shoes.dmi'
 	icon_state = "dwarf"
-	inhand_icon_state = "dwarf"
+	inhand_icon_state = "jackboots"
 	body_parts_covered = FEET
 	species_exception = list(/datum/species/dwarf)
+
+//Forged Armour
+/obj/item/clothing/suit/armor/vest/dwarf
+	name = "dwarven armour"
+	desc = "Great for stopping sponges."
+	worn_icon = 'russstation/icons/mob/suit.dmi'
+	icon = 'russstation/icons/obj/clothing/suits.dmi'
+	icon_state = "dwarf"
+	inhand_icon_state = "swat_suit"
+	body_parts_covered = CHEST|GROIN|ARMS
+	cold_protection = CHEST|GROIN|ARMS
+	heat_protection = CHEST|GROIN|ARMS
+	armor = list(melee = 50, bullet = 10, laser = 10, energy = 10, bomb = 0, bio = 0, rad = 0, fire = 80, acid = 80)
+	strip_delay = 80
+	equip_delay_self = 60
+	species_exception = list(/datum/species/dwarf)
+
+/obj/item/clothing/suit/armor/vest/dwarf/CheckParts(list/parts_list)
+	..()
+	var/obj/item/mold_result/armour_plating/S = locate() in contents
+	if(S)
+		var/image/Q = image(icon, icon_state)
+		Q.color = S.color
+		add_overlay(Q)
+		smelted_material = new S.smelted_material.type()
+		name = "[S.material_type] armour"
+		desc = "Armour forged from [S.material_type]."
+		var/list/defenses = armor
+		for(var/A in defenses)
+			A = S.attack_amt/100
```

Comment:
Will this actually mutate the corresponding value in the list or will this just overwrite the loop variable like in most languages?

Reply:
Fixed this issue. Check lines 41 to 113 in dwarf_clothes.dm

Reply:
I noticed before you were dividing attack_amt by 100, but in the new version, you're not? Just wanted to make sure that's intentional